File: refactoring-practice-1/src/main/java/dojo/hide_delegate/Person.java
package dojo.hide_delegate;

/*
 * Hozzunk létre egy getManager metódust, ami visszaadja a department.manager értékét,
 * így elrejthető a department mező.
 */
public class Person {

	private Department department;
	
	public Department getDepartment() {
		return department;
	}
	
	public void setDepartment(Department department) {
		this.department = department;
	}
}

File: refactoring-practice-1/src/main/typescript/hide_delegate/person.ts
/*
 * Hozzunk létre egy getManager metódust, ami visszaadja a department.manager értékét,
 * így elrejthető a department mező.
 */
import {Department} from "./department.js";

export class Person {

    private department: Department;

    public getDepartment() {
        return this.department;
    }

    public setDepartment(department: Department) {
        this.department = department;
    }
}
